logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+ * @param {Accounts} accounts
+ */
+TransactionPool.prototype.bind = function (accounts) {
+	modules = {
+		accounts: accounts,
+	};
+};
+
+/**
+ * Gets invalid, unverified, verified.pending and verified.ready counters.
+ * @implements {__private.countTransactionsPool}
+ * @return {Object} unverified, pending, ready
+ */
+TransactionPool.prototype.getUsage = function () {
+	return {
+		unverified: pool.unverified.count,
+		pending: pool.verified.pending.count,
+		ready: pool.verified.ready.count,
+		invalid: pool.invalid.count,
+		total: __private.countTransactionsPool()
+	};
+};
+
+/**
+ * Gets transaction based on transaction id.
+ * Checks all pool lists: unverified, pending, ready.
+ * @param {string} id
+ * @return {Object} transaction, status
+ */
+TransactionPool.prototype.get = function (id) {
+	var transaction = pool.unverified.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'unverified'
+		};
+	}
+
+	transaction = pool.verified.pending.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'pending'
+		};
+	}
+	
+	transaction = pool.verified.ready.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'ready'
+		};
+	}
+	return {
+		transaction: undefined,
+		status: 'Transaction not in pool'
+	};
+};
+
+/**
+ * Gets all transactions based on limited filters.
+ * @implements {__private.getTransactionsFromPoolList}
+ * @implements {__private.getAllPoolTransactionsByFilter}
+ * @param {string} filter
+ * @param {Object} params
+ * @return {[transaction]} transaction list based on filter.
+ */
+TransactionPool.prototype.getAll  = function (filter, params) {
+	switch (filter) {
+		case 'unverified':
+			return __private.getTransactionsFromPoolList(pool.unverified.transactions, params.reverse, params.limit);
+		case 'pending':
+			return __private.getTransactionsFromPoolList(pool.verified.pending.transactions, params.reverse, params.limit);
+		case 'ready':
+			return __private.getTransactionsFromPoolList(pool.verified.ready.transactions, params.reverse, params.limit);
+		case 'sender_id':
+			return __private.getAllPoolTransactionsByFilter({'senderId': params.id});
+		case 'sender_pk':
+			return __private.getAllPoolTransactionsByFilter({'senderPublicKey': params.publicKey});
+		case 'recipient_id':
+			return __private.getAllPoolTransactionsByFilter({'recipientId': params.id});
+		case 'recipient_pk':
+			return __private.getAllPoolTransactionsByFilter({'requesterPublicKey': params.publicKey});
+		default:
+			return 'Invalid filter';
+	}
+};
+
+/**
+ * Gets ready transactions ordered by fee and received time.
+ * @param {number} limit
+ * @return {transaction[]}
+ */
+TransactionPool.prototype.getReady = function (limit) {
+	var r = _.orderBy(pool.verified.ready.transactions, ['fee', 'receivedAt'],['desc', 'asc']);
+	if (limit && limit < r.length) {
```

Comment:
Is `limit < r.length` needed?

Reply:
to avoid splice if we don't receive a limit or we don't have enough transactions